src/raftstore/store/peer.rs

```diff
@@ -979,14 +991,32 @@ impl Peer {
                     }
                 }
 
-                // We only care about split commands that are committed
-                // in the current term.
-                if split_to_be_updated && entry.term == self.term() {
+                // We care about split/merge commands that are committed in the current term.
+                if entry.term == self.term() && (split_to_be_updated || merge_to_be_update) {
                     let ctx = ProposalContext::from_bytes(&entry.context);
-                    if ctx.contains(ProposalContext::SPLIT) {
+                    if split_to_be_updated && ctx.contains(ProposalContext::SPLIT) {
                         self.last_committed_split_idx = entry.index;
                         split_to_be_updated = false;
                     }
+                    if merge_to_be_update {
+                        if ctx.contains(ProposalContext::ROLLBACK_MERGE) {
+                            // We committed rollback merge, in order to enable read index,
+                            // we need to unset the pending_merge.
+                            //
+                            // Also local read is safe now, we renew its leader lease
+                            // as soon as possible, in the current implementation `apply_reads`
+                            // may renew the lease if there are any consistent reads.
+                            self.pending_merge = false;
+                            merge_to_be_update = false;
+                        } else if ctx.contains(ProposalContext::PREPARE_MERGE) {
```

Comment:
Should we care about COMMIT_MERGE here?

Reply:
No, we dont need to, COMMIT_MERGE is committed a target region, so it wont appear here.

Reply:
How do you know it's definitely going to be applied?